pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+	tektonhubconciler "github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonhub"
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+)
+
+// Reconciler implements controller.Reconciler for TektonHub resources.
+type Reconciler struct {
+	// kubeClientSet allows us to talk to the k8s for core APIs
+	kubeClientSet kubernetes.Interface
+	// operatorClientSet allows us to configure operator objects
+	operatorClientSet clientset.Interface
+	// manifest is empty, but with a valid client and logger. all
+	// manifests are immutable, and any created during reconcile are
+	// expected to be appended to this one, obviating the passing of
+	// client & logger
+	manifest mf.Manifest
+	// Platform-specific behavior to affect the transform
+	extension common.Extension
+	// enqueueAfter enqueues a obj after a duration
+	enqueueAfter func(obj interface{}, after time.Duration)
+}
+
+var (
+	apiConfigMapName string = "api"
+	errKeyMissing    error  = fmt.Errorf("secret doesn't contains all the keys")
+	namespace        string = ""
+	// Check that our Reconciler implements controller.Reconciler
+	_ tektonhubconciler.Interface = (*Reconciler)(nil)
+	_ tektonhubconciler.Finalizer = (*Reconciler)(nil)
+)
+
+const (
+	dbInstallerSet          = "DbInstallerSet"
+	dbMigrationInstallerSet = "DbMigrationInstallerSet"
+	apiInstallerSet         = "ApiInstallerSet"
+
+	releaseVersionKey  = "operator.tekton.dev/release-version"
+	createdByKey       = "operator.tekton.dev/created-by"
+	targetNamespaceKey = "operator.tekton.dev/target-namespace"
+	createdByValue     = "TektonHub"
+)
+
+// FinalizeKind removes all resources after deletion of a TektonHub.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonHub) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	installerSets := original.Status.HubInstallerSet
+	if len(installerSets) == 0 {
+		return nil
+	}
+
+	for _, value := range installerSets {
+		err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().Delete(ctx, value, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	// List all TektonHub to determine if cluster-scoped resources should be deleted.
+	ths, err := r.operatorClientSet.OperatorV1alpha1().TektonHubs().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all TektonHubs: %w", err)
+	}
+
+	for _, th := range ths.Items {
+		if th.GetDeletionTimestamp().IsZero() {
+			// Not deleting all TektonHubs. Nothing to do here.
+			return nil
+		}
+	}
+
+	if err := r.extension.Finalize(ctx, original); err != nil {
+		logger.Error("Failed to finalize platform resources", err)
+	}
+	logger.Info("Deleting cluster-scoped resources")
+	manifest, err := r.installed(ctx, original)
```

Comment:
is this necessary, as we are creating all resources throught installersets, deleting installersets should delete the resources. In addition, leave as much as clusterOperations to TektonInstallerSet Reconciler.